splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSet.java

```diff
@@ -591,7 +592,7 @@ public String getAttribute(String name) {
                     ValueRow vr = (ValueRow) iterator.next();
                     context.recordWrite();
 
-                    rw.write(null, encoder.toRow(vr));
+                    rw.write(null, ParquetWriterService.getFactory().encodeToRow(tableSchema, vr, encoder));
```

Comment:
for other reviewers: underlying change here is that Spark2.4 uses `encoder.toRow(valueRow);` , while Spark3.0 uses `return encoder.createSerializer().apply(valueRow);` . see `hbase_sql/orc-spark-3.0/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java` and 2.4